test/val/val_tensor_test.cpp

```diff
@@ -0,0 +1,1203 @@
+             %uint = OpTypeInt 32 0
+            %float = OpTypeFloat 32
+             %fnty = OpTypeFunction %void
+        %uint_vec4 = OpTypeVector %uint 4
+           %uint_0 = OpConstant %uint 0
+      %uint_0_spec = OpSpecConstant %uint 0
+           %uint_1 = OpConstant %uint 1
+           %uint_2 = OpConstant %uint 2
+           %uint_3 = OpConstant %uint 3
+           %uint_4 = OpConstant %uint 4
+          %uint_42 = OpConstant %uint 42
+       %uint_1_bis = OpConstant %uint 1
+          %float_1 = OpConstant %float 1
+%uint_vec4_1_1_1_1 = OpConstantComposite %uint_vec4 %uint_1 %uint_1 %uint_1 %uint_1
+        %uint_arr1 = OpTypeArray %uint %uint_1
+        %uint_arr2 = OpTypeArray %uint %uint_2
+        %uint_arr3 = OpTypeArray %uint %uint_3
+        %uint_arr4 = OpTypeArray %uint %uint_4
+       %float_arr4 = OpTypeArray %float %uint_4
+%uint_arr4_1_1_1_1 = OpConstantComposite %uint_arr4 %uint_1 %uint_1 %uint_1 %uint_1
+%uint_arr4_0_0_0_0_spec = OpSpecConstantComposite %uint_arr4 %uint_0_spec %uint_0_spec %uint_0_spec %uint_0_spec
+    %uint_arr2_1_1 = OpConstantComposite %uint_arr2 %uint_1 %uint_1
+%float_arr4_1_1_1_1 = OpConstantComposite %float_arr4 %float_1 %float_1 %float_1 %float_1
+%uint_arr4_1_1_0_1 = OpConstantComposite %uint_arr4 %uint_1 %uint_1 %uint_0 %uint_1
+ %uint_ptr_Private = OpTypePointer Private %uint
+%uint_arr4_ptr_Private = OpTypePointer Private %uint_arr4
+       %uint_var_1 = OpVariable %uint_ptr_Private Private %uint_1
+%var_uint_arr4_1_1_1_1 = OpVariable %uint_arr4_ptr_Private Private %uint_arr4_1_1_1_1
+    %tensor_uint_4 = OpTypeTensorARM %uint %uint_4
+     %tensor_float = OpTypeTensorARM %float
+%tensor_uint_4_ptr_UniformConstant = OpTypePointer UniformConstant %tensor_uint_4
+%tensor_var = OpVariable %tensor_uint_4_ptr_UniformConstant UniformConstant
+%tensor_float_ptr_UniformConstant = OpTypePointer UniformConstant %tensor_float
+%tensor_var_float_unranked = OpVariable %tensor_float_ptr_UniformConstant UniformConstant
+)";
+  const std::string footer = R"(
+             %fnep = OpFunction %void None %fnty
+            %label = OpLabel
+                     OpReturn
+                     OpFunctionEnd
+)";
+  return header + body + footer;
+}
+
+//
+// Type tests
+//
+
+TEST_F(ValidateTensor, ValidTypeElementTypeOnly) {
+  const std::string src = R"(
+    %test_type = OpTypeTensorARM %uint
+)";
+  std::string spvasm = GenerateModule(src);
+  CompileSuccessfully(spvasm, SPVENV);
+  EXPECT_EQ(SPV_SUCCESS, ValidateInstructions(SPVENV));
+}
+
+TEST_F(ValidateTensor, InvalidTypeTensorElementTypeNotScalar) {
+  const std::string src = R"(
+    %test_type = OpTypeTensorARM %void
+)";
+  std::string spvasm = GenerateModule(src);
+  CompileSuccessfully(spvasm, SPVENV);
+  EXPECT_EQ(SPV_ERROR_INVALID_ID, ValidateInstructions(SPVENV));
+  EXPECT_THAT(
+      getDiagnosticString(),
+      MatchesRE2Regex("OpTypeTensorARM Element Type <id> "
+                      "'[[:digit:]]\\[%void\\]' is not a scalar type.*"));
```

Comment:
The actual string is:
```
error: 46: OpTypeTensorARM Element Type <id> '4[%void]' is not a scalar type.
  %35 = OpTypeTensorARM %void
```

Suggest changing this to  `'.*\\[%void\\]' is not a scalar`

The point is we get the message we thought we were getting.

And similarly for other uses of `[[:digit:]]`.

And if the suggestion fails again because of square brackets, use `.` more liberally.  But I've seen other existing MatchRegex match literal `[`

Reply:
>I've seen other existing MatchRegex match literal [

AFAIU, the literal `[` was not the issue. It was the `[` used in character classes.